lib/csn-runtime-extension.js

```diff
@@ -0,0 +1,24 @@
+const cds = require('@sap/cds')
+const { LinkedDefinitions } = require("@sap/cds/lib/core/linked-csn")
+
+Object.defineProperty(cds.builtin.classes.entity.prototype, '_attachments', {
+  get() {
+    const entity = this;
+    return {
+      get hasAttachmentsComposition() {
+        return entity.compositions && Object.keys(entity.compositions).some(ele => entity.compositions[ele]._target?.["@_is_media_data"] || entity.compositions[ele]._target?._attachments.hasAttachmentsComposition)
```

Comment:
```suggestion
        return entity.compositions && Object.keys(entity.compositions).some(ele => entity.compositions[ele]._target?.["@_is_media_data"] || entity.compositions[ele]._target?._attachments?.hasAttachmentsComposition)
```

not 100% sure but this might fail if _attachments is undefined?

Reply:
When a target exists _attachments would never be undefined.